src/main/java/org/assertj/core/api/Assertions.java

```diff
@@ -1104,6 +1104,31 @@ public static AtomicLongArrayAssert assertThat(AtomicLongArray actual) {
     return AssertionsForClassTypes.assertThatCode(shouldRaiseOrNotThrowable);
   }
 
+  /**
+   * Creates a new instance of <code>{@link ObjectAssert}</code> for any object.
+   *
+   * <p>
+   * This overload is useful, when an overloaded method of assertThat(...) takes precedence over the generic {@link #assertThat(Object)}.
+   * </p>
+   *
+   * Example :
+   * <pre><code class='java'>class MyList<T> extends AbstractList<T> {
+   *     public int allocationSize() { ... }
+   * }
+   *
+   * // cast necessary because {@link #assertThat(List)} "forgets" actual type
+   * assertThat(new MyList<Object>()).matches(list -> ((MyList<Object>) list).allocationSize() == 13);
+   * // no cast needed, but also no additional list assertions
+   * assertThatObject(new MyList<Object>()).matches(list -> list.allocationSize() == 13);</code></pre>
+   *
+   * @param <T> the type of the actual value.
+   * @param actual the actual value.
+   * @return the created assertion object.
+   */
+  public static <T> ObjectAssert<T> assertThatObject(T actual) {
```

Comment:
this also needs to be added to `WithAssertion` and `BDDAssertions` but for the latter as `thenObject`